src/engine/SCons/Tool/msvs.py

```diff
@@ -578,11 +578,8 @@ def AddConfig(self, variant, buildtarget, outdir, runfile, cmdargs, cppdefines,
         for i in range(len(variants)):
             AddConfig(self, variants[i], buildtarget[i], outdir[i], runfile[i], cmdargs[i], cppdefines[i], cpppaths[i])
 
-        self.platforms = []
-        for key in list(self.configs.keys()):
-            platform = self.configs[key].platform
-            if platform not in self.platforms:
-                self.platforms.append(platform)
+        self.platforms = {p.platform for p in self.configs.values()}
```

Comment:
Is order important here?

Reply:
I don't think so, it's only used, as far as I can tell, in writing text to the msvc project file. In any case, the set comprehension will get things in the same order as the list creation did, since both get there by looping over the entries fetched from the self.configs dict.

Reply:
Set insertion order isn't maintained until python 3.7?  If the contents of the msvc project file are a source for anything and the order changes, then there'll be extraneous rebuilds.  maybe 
```
# TODO when floor moved to py37 remove list
self.platforms =list( {p.platform for p in self.configs.values()})
```

Reply:
That doesn't help... if the order it came out of `self.configs` is important to keep then using a set for uniqueness can't be used - and converting the set to the list just preserves what the set looked like. (oh, and there is no version of Python where set order is "preserved", as people keep trying to point out, that's a meaningless concept in mathematics). Need to look if there's any reason the order would matter.

Reply:
If you think it matters, and I'm not sure it does, I'll rewrite it this way, which at least guarantees consistency from run to run:
```
self.platforms =sorted({p.platform for p in self.configs.values()})
```

Reply:
How about something like this with a note that set can be used when python baseline is bumped to py37?
```def unique(sequence):
    seen = set()
    return [p.platform for p in self.configs.values() if not (p.plaform in seen or seen.add(p.platform))]```

Reply:
sure, that will work.  maybe it's cheaper just to do it than to sort out whether it's necessary? in other words... does the original order matter (order-preserving)? or is it sufficient that the same data gives the same result from run to run (sorting)? or does either not matter (not-order-preserving)

Reply:
Updated, though at least in running the tests, some debug shows there is only ever one value ('win32').... maybe we don't test rigorously enough?